libsolidity/codegen/ContractCompiler.cpp

```diff
@@ -971,31 +971,46 @@ bool ContractCompiler::visit(TryStatement const& _tryStatement)
 void ContractCompiler::handleCatch(vector<ASTPointer<TryCatchClause>> const& _catchClauses)
 {
 	// Stack is empty.
-	ASTPointer<TryCatchClause> structured{};
+	ASTPointer<TryCatchClause> error{};
+	ASTPointer<TryCatchClause> panic{};
 	ASTPointer<TryCatchClause> fallback{};
 	for (size_t i = 1; i < _catchClauses.size(); ++i)
 		if (_catchClauses[i]->errorName() == "Error")
-			structured = _catchClauses[i];
+			error = _catchClauses[i];
+		else if (_catchClauses[i]->errorName() == "Panic")
+			panic = _catchClauses[i];
 		else if (_catchClauses[i]->errorName().empty())
 			fallback = _catchClauses[i];
 		else
 			solAssert(false, "");
 
-	solAssert(_catchClauses.size() == 1ul + (structured ? 1 : 0) + (fallback ? 1 : 0), "");
+	solAssert(_catchClauses.size() == 1ul + (error ? 1 : 0) + (panic ? 1 : 0) + (fallback ? 1 : 0), "");
 
 	evmasm::AssemblyItem endTag = m_context.newTag();
 	evmasm::AssemblyItem fallbackTag = m_context.newTag();
-	if (structured)
+	evmasm::AssemblyItem panicTag = m_context.newTag();
+	if (error != nullptr || panic != nullptr)
```

Comment:
```suggestion
	if (error || panic)
```

Reply:
It was weirdly failing before I changed it, I'll try to revert again.

Reply:
Malformed in the sense that the error data is too short for the length stored inside it.